### PR TITLE
Do not attempt to detect container runtime in native-sources build

### LINKED ISSUE #33161: Quarkus 3 quarkus.package.type=native-sources require docker/podman or native-image
### Describe the bug

Starting from version 3 of Quarkus, the behavior of the command './gradlew build '-Dquarkus.package.type=native-sources'' changed, as it now requires either Docker/Podman or native-image installed on the host. Up until version 2.16.0.Final, it did not require Docker or native-image to be installed.

### Expected behavior

It is expected not to require Docker/Podman or native-image, just as it worked in version 2.16.

### Actual behavior
```
 ./gradlew build -Dquarkus.package.type=native-sources                                                       ─╯

> Configure project :
WARNING: The task 'testCodeCoverageReport' of type 'JacocoReport' does not follow the naming convention "jacoco(TestTask)Report", where "TestTask" is the capitalized name of the test task covered by the report. We improvise by using 'jacocoLogCoverage13' as a fallback task name.
WARNING: The task 'integrationTestsCodeCoverageReport' of type 'JacocoReport' does not follow the naming convention "jacoco(TestTask)Report", where "TestTask" is the capitalized name of the test task covered by the report. We improvise by using 'jacocoLogCoverage14' as a fallback task name.
Cannot find the `native-image` in the GRAALVM_HOME, JAVA_HOME and System PATH. Install it using `gu install native-image` Attempting to fall back to container build.

> Task :quarkusBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusBuild'.
> There was a failure while executing work items
   > A failure occurred while executing io.quarkus.gradle.tasks.worker.BuildWorker
      > io.quarkus.builder.BuildException: Build failure: Build failed due to errors
                [error]: Build step io.quarkus.deployment.pkg.steps.NativeImageBuildStep#resolveNativeImageBuildRunner threw an exception: java.lang.IllegalStateException: No container runtime was found. Make sure you have either Docker or Podman installed in your environment.
                at io.quarkus.runtime.util.ContainerRuntimeUtil.detectContainerRuntime(ContainerRuntimeUtil.java:99)
                at io.quarkus.runtime.util.ContainerRuntimeUtil.detectContainerRuntime(ContainerRuntimeUtil.java:41)
                at java.base/java.util.Optional.orElseGet(Optional.java:364)
                at io.quarkus.deployment.pkg.steps.NativeImageBuildContainerRunner.<init>(NativeImageBuildContainerRunner.java:31)
                at io.quarkus.deployment.pkg.steps.NativeImageBuildLocalContainerRunner.<init>(NativeImageBuildLocalContainerRunner.java:21)
                at io.quarkus.deployment.pkg.steps.NativeImageBuildStep.resolveNativeImageBuildRunner(NativeImageBuildStep.java:332)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
                at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.base/java.lang.reflect.Method.invoke(Method.java:568)
                at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
                at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
                at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
                at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
                at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
                at java.base/java.lang.Thread.run(Thread.java:833)
                at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```
### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gardlew 8

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by zakkak**
Closes: #33161

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package io.quarkus.deployment.pkg.steps;
+
+import java.nio.file.Path;
+import java.util.List;
+
+public class NativeImageBuildRunnerDummy extends NativeImageBuildRunner {
```
[the remark this answers is not in the document]
Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java:6]: > Let's come up with a better name please :).

Can you please indicate what's wrong with this name?

`Constant` is not a good name. We need to somehow indicate that it is a dummy, i.e. not meant to be actually used to build a native image. WDYT about `Phony`?

[the remark this answers is not in the document]
Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java:6]: Maybe `NoopNativeImageBuildRunner` or `ThrowingNativeImageBuildRunner`?

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    private static final String MESSAGE = "NativeImageBuildRunnerDummy is note meant to be used to perform an actual build.";
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java:8]:
```suggestion
    private static final String MESSAGE = "NativeImageBuildRunnerDummy is not meant to be used to perform an actual build.";
```

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    @Override
+    public boolean isContainer() {
+        return isContainer;
+    }
+
+    @Override
+    protected String[] getGraalVMVersionCommand(List<String> args) {
+        throw new RuntimeException(MESSAGE);
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java:22]: Please use `UnsupportedOperationException`.

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 4]
```diff
@@ -786,10 +786,11 @@ $ ./mvnw clean package -Dquarkus.package.type=native-sources -Dquarkus.native.co
 cd target/native-sources
 docker run \
   -it \
+  --user $(id -ur):$(id -gr) \
   --rm \
-  --v $(pwd):/work \# <1>
+  -v $(pwd):/work \# <1>
   -w /work \# <2>
-  --entrypoint bin/sh \
+  --entrypoint /bin/sh \
```
Comment (gsmet) [inline docs/src/main/asciidoc/building-native-image.adoc:793]: Is it related to this change?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a few comments.

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java:6]: I prefer both

[the remark this answers is not in the document]
Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRunnerDummy.java:6]: I like `NoopNativeImageBuildRunner` :+1:

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 4 above)
Comment (zakkak) [reply docs/src/main/asciidoc/building-native-image.adoc:793]: Yes, it's from the documentation part talking about using `native-sources`. Testing this PR requires using that command.

**Comment by zakkak** [comment]
The CI failure seems unrelated to me.

**Review by gsmet — APPROVED** [review]

**Pull request merged**